langroid/agent/chat_agent.py

```diff
@@ -646,7 +794,21 @@ def llm_response(
             else (message.oai_tool_choice if message is not None else "auto")
         )
         with StreamingIfAllowed(self.llm, self.llm.get_stream()):
-            response = self.llm_response_messages(hist, output_len, tool_choice)
+            try:
+                response = self.llm_response_messages(hist, output_len, tool_choice)
+            except openai.BadRequestError as e:
+                if self.any_strict:
+                    self.disable_strict = True
```

Comment:
As OpenAI doesn't support the full JSON schema, it is possible that enabling strict output (which we do automatically) could cause errors; this recovery mechanism is designed to catch them. I'm not aware of precise specifications of the subset of JSON schema supported, but if they exist it would be much better to validate the schemas directly.

We could detect which tools are triggering the failures from the error messages (which contain the responsible tool name) but this may be brittle.